micropip/package_index.py

```diff
@@ -123,11 +131,42 @@ def _parse_pep691_response(
                 version = parse_version(filename)
             except (InvalidVersion, InvalidWheelFilename):
                 continue
+            if file["url"].startswith("/"):
+                file["url"] = index_base_url + file["url"]
 
             releases[version].append(file)
 
         return name, releases
 
+    @classmethod
+    def _compatible_wheels(
+        cls, files: list[dict[str, Any]], version: Version, name: str
+    ) -> Generator[WheelInfo, None, None]:
+        for file in files:
+            filename = file["filename"]
+
+            # Checking compatibility takes a bit of time,
+            # so we use a generator to avoid doing it for all files.
+            compatible = is_package_compatible(filename)
+            if not compatible:
+                continue
+
+            # JSON API has a "digests" key, while Simple API has a "hashes" key.
+            hashes = file["digests"] if "digests" in file else file["hashes"]
+            sha256 = hashes.get("sha256")
+
+            # Size of the file in bytes, if available (PEP 700)
+            # This key is not available in the Simple API HTML response, so this field may be None
+            size = file.get("size")
+            yield WheelInfo.from_package_index(
+                name=name,
+                filename=filename,
+                url=file["url"],
+                version=version,
+                sha256=sha256,
+                size=size,
+            )
```

Comment:
This just avoids a closure by making `name` a parameter, and making things a bit more explicit.